toolkit/hooks/src/use-progress.ts

```diff
@@ -78,10 +80,15 @@ export let useProgress = ({
   }, [running]);
 
   useEffect(() => {
-    if (progress < 100 && totalFileSize !== 0) {
+    if (progress < 100 && totalFileSize !== 0 && progressLoop === false ) {
```

Comment:
Yes, it Possible to do that with the logical or but i could not understand why we need to do that there is any specific reason i could not understand because both are doing same work and logic also same for both.

### First:
```
if (progress < 100 && totalFileSize !== 0 && progressLoop === false ) {
      clearInterval(intervalRef.current as NodeJS.Timeout);
      start();
    }
    else if (totalFileSize !== 0 && progressLoop === true ) {
      clearInterval(intervalRef.current as NodeJS.Timeout);
      start();
    }
```

### Second:
```
if (progress < 100 && totalFileSize !== 0 && progressLoop === false || totalFileSize !== 0 && progressLoop === true) {
      clearInterval(intervalRef.current as NodeJS.Timeout);
      start();
    }
```

Reply:
To prevent code repatations. Here you are doing same thing on two different conditions. This should be avoided.

Reply:
ok, Removed Second Case is implemented.